segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/BadSegmentTypeException.java

```diff
@@ -19,21 +19,21 @@
  * Exception that is thrown whenever a segment of the wrong type is accessed (i.e., we want a StreamSegment but were given
  * the name of a Table Segment).
  */
-public class BadSegmentTypeException extends StreamSegmentException {
+public class BadSegmentTypeException extends IllegalArgumentException {
```

Comment:
What are the side effects of this to other components?

Reply:
This exception has been there for 3 years but never used anywhere. I have made it extend `IllegalArgumentException` and replaced a single place in the code throwing `IllegalArgumentException` with it. So any existing code will still work as it did, but now we have a specific exception to tell us when we try to access the wrong segment type.